invertedai/cosimulation.py

```diff
@@ -33,30 +33,45 @@ def __init__(
         location: str,
         ego_agent_mask: Optional[List[bool]] = None,
         monitor_infractions: bool = False,
-        render_birdview: bool = False,
+        get_birdview: bool = False,
         # sufficient arguments to initialize must also be included
         **kwargs,
     ):
         self._location = location
-        response = initialize(location=location, **kwargs)
+        response = initialize(
+            location=location,
+            get_birdview=get_birdview,
+            get_infractions=monitor_infractions,
+            **kwargs,
+        )
+        if monitor_infractions and (response.infractions is not None):
+            # TODO: package infractions in dataclass
+            self._infractions = (
+                [inf.collisions for inf in response.infractions],
+                [inf.offroad for inf in response.infractions],
+                [inf.wrong_way for inf in response.infractions],
+            )
+        else:
+            self._infractions = None
         self._agent_count = len(
             response.agent_attributes
         )  # initialize may produce different agent count
         self._agent_attributes = response.agent_attributes
         self._agent_states = response.agent_states
         self._recurrent_states = response.recurrent_states
         self._monitor_infractions = monitor_infractions
-        self._infractions = None
-        self._render_birdview = render_birdview
+        self._render_birdview = get_birdview
         self._birdview = None
+        if self._render_birdview:
+            self._birdview = response.birdview
         if ego_agent_mask is None:
             self._ego_agent_mask = [False] * self._agent_count
         else:
             self._ego_agent_mask = ego_agent_mask[: self._agent_count]
         # initialize might not return the exact number of agents requested,
         # in which case we need to adjust the ego agent mask
         if len(self._ego_agent_mask) > self._agent_count:
-            self._ego_agent_mask = self._ego_agent_mask[:self._agent_count]
+            self._ego_agent_mask = self._ego_agent_mask[: self._agent_count]
```

Comment:
Why the extra space?

Reply:
I had the black code formatter on.
To be consistent with you and Ruishen, I will use pep8 from now on and I guess you have max-line-length=119 and not 79.